compiler/types.nim

```diff
@@ -502,7 +502,10 @@ proc typeToString(typ: PType, prefer: TPreferedDesc = preferName): string =
     #internalAssert t.len == 0
     result = "untyped"
   of tyFromExpr:
-    result = renderTree(t.n)
+    if t.n == nil:
+      result = "unknown"
+    else:
+      result = "type(" & renderTree(t.n) & ")"
```

Comment:
So ... ``tyFromExpr`` is yet another subtle variant of ``tyTypeDesc``? 

Can you explain the difference between ``tyInferred``, ``tyFromExpr`` and ``tyStatic`` to a fellow compiler developer of your choice without making him eyeroll?

Reply:
Let's do this over chat. I though you've learned what `tyFromExpr` is by now.

Reply:
I thought I would have learned but I doubt it now. But that's not the point. The point is to at least document this better in the source code, not via chatting with me.